File: import-quotes.js
// import-quotes.js
const { createClient } = require('@supabase/supabase-js');
const fs = require('fs');

// GANTI NILAI DI BAWAH INI DENGAN DATA DARI DASHBOARD SUPABASE ANDA
const SUPABASE_URL = 'https://ipvookfvxbalfxusomuc.supabase.co'; // Harus diawali https://
const SUPABASE_SERVICE_ROLE_KEY = 'sb_publishable_7pG1fGeeg0J7iGBDF_eO2w_Vrh5TLoF';

const supabase = createClient(SUPABASE_URL, SUPABASE_SERVICE_ROLE_KEY);

async function bulkImport() {
  try {
    // Pastikan file quotes.json ada di folder yang sama
    const rawData = fs.readFileSync('quotes.json', 'utf8');
    const quotesData = JSON.parse(rawData);

    console.log(`🚀 Memulai impor ${quotesData.length} kata motivasi...`);

    const chunkSize = 500;
    for (let i = 0; i < quotesData.length; i += chunkSize) {
      const chunk = quotesData.slice(i, i + chunkSize);
      
      const { error } = await supabase
        .from('motivational_quotes')
        .insert(chunk);

      if (error) {
        console.error(`❌ Gagal pada baris ke-${i}:`, error.message);
      } else {
        console.log(`✅ Berhasil memasukkan ${i + chunk.length} data...`);
      }
    }

    console.log('\n✨ Semua data berhasil dimasukkan ke database!');
  } catch (err) {
    console.error('❌ Terjadi kesalahan script:', err.message);
  }
}

bulkImport();

File: import_quotes.py
# import_quotes.py
from supabase import create_client, Client

url: str = "https://ipvookfvxbalfxusomuc.supabase.co"
key: str = "sb_secret_0Ch9zNGXH6j9VaFX5z-qLw_uiYwJork"
supabase: Client = create_client(url, key)

# Anda bisa memuat data dari file CSV menggunakan pandas
# import pandas as pd
# df = pd.read_csv('quotes.csv')
# quotes_data = df.to_dict('records')

quotes_data = [
    {"category": "pagi", "quote": "Hari baru, semangat baru untuk Bandarin!"},
    {"category": "siang", "quote": "Fokus adalah kunci efisiensi."},
    # ... ribuan data lainnya
]

def chunk_list(data, size):
    for i in range(0, len(data), size):
        yield data[i:i + size]

def bulk_import():
    print(f"Mengimpor {len(quotes_data)} data...")
    
    # Batch insert 500 baris per request
    for chunk in chunk_list(quotes_data, 500):
        try:
            response = supabase.table("motivational_quotes").insert(chunk).execute()
            print(f"Batch berhasil diimpor.")
        except Exception as e:
            print(f"Terjadi kesalahan: {e}")

if __name__ == "__main__":
    bulk_import()